pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/util/PackageSettings.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.util;
+
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Interface for package specific settings.
+ *
+ * The settings needed for a specific package to build properly
+ * or to customize its build can be obtained from the API provided
+ * by this interface.
+ *
+ * @param <T> Type of the object that represents package information.
+ */
+public interface PackageSettings<T> {
```

Comment:
Do we want to bound T to be something?

Reply:
The intent is to have a generic type for future development. No need to change T in this interface when we change the package information type.